cli/register/register.go

```diff
@@ -629,6 +629,17 @@ func verifyRegisterParamters(org, pattern, nodeOrgFromFlag string, patternFromFl
 		}
 	}
 
+	if nodeId, _ := cliutils.SplitIdToken(nodeIdTok); nodeId != "" {
```

Comment:
Should also consider getting node id from the agent if the nodeId is empty. See line 129-135. Agent gets the node id from file /etc/default/horizon. User could put a bad id there.

Reply:
Added `nodeId` fetching from agent in case of empty `nodeId`